enclave/core/tracee.c

```diff
@@ -69,6 +77,63 @@ void oe_log_init_ecall(const char* enclave_path, uint32_t log_level)
     }
 }
 
+static void* oe_enclave_log_context = NULL;
+static oe_enclave_log_callback_t oe_enclave_log_callback = NULL;
+
+oe_result_t oe_enclave_log_set_callback(
+    void* context,
+    oe_enclave_log_callback_t callback)
+{
+    oe_result_t result = OE_OK;
+
+    OE_CHECK_NO_TRACE(oe_mutex_lock(&_log_lock));
+
+    oe_enclave_log_context = context;
+    oe_enclave_log_callback = callback;
+    oe_mutex_unlock(&_log_lock);
+
+    OE_TRACE_INFO("enclave logging callback is set");
```

Comment:
Is this trace safe given that it is part of logging/tracing itself.
If yes, add a comment that it does not introduce a recursive call.

Reply:
since we call this after the basic logging functions are already setup, as a function call from the enclave application itself, this trace is safe.. I'll add a comment to the source.